catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergCatalogOperations.java

```diff
@@ -495,13 +495,21 @@ public Table createTable(
       IcebergColumn[] icebergColumns =
           Arrays.stream(columns)
               .map(
-                  column ->
-                      new IcebergColumn.Builder()
-                          .withName(column.name())
-                          .withType(column.dataType())
-                          .withComment(column.comment())
-                          .withNullable(column.nullable())
-                          .build())
+                  column -> {
+                    if (!column.defaultValue().equals(Column.DEFAULT_VALUE_NOT_SET)) {
+                      // Iceberg column default value is WIP, see
+                      // https://github.com/apache/iceberg/pull/4525
+                      throw new IllegalArgumentException(
+                          "Iceberg does not support column default value. Illegal column: "
+                              + column.name());
+                    }
+                    return new IcebergColumn.Builder()
+                        .withName(column.name())
+                        .withType(column.dataType())
+                        .withComment(column.comment())
+                        .withNullable(column.nullable())
```

Comment:
pass `Column.DEFAULT_VALUE_NOT_SET` to IcebergColumn? then Iceberg no need to handle null values.

Reply:
It's already handled in `com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergColumn.Builder#internalBuild`

Reply:
I think it's more clear to pass Column.DEFAULT_VALUE_NOT_SET here, then no need to do extra logic in `IcebergColumn.Builder#internalBuild`

Reply:
This means we need to make this change wherever the `IcebergColumn.Builder` is being used. But the logic added in `IcebergColumn.Builder#internalBuild` only needs to be added once.